core/rawdb/accessors_chain.go

```diff
@@ -1041,17 +1044,20 @@ func PruneBlocks(tx kv.RwTx, blockTo uint64, blocksDeleteLimit int) error {
 
 	for k, _, err := c.Current(); k != nil; k, _, err = c.Next() {
 		if err != nil {
-			return err
+			return existBlocksToPrune, err
 		}
 
 		n := binary.BigEndian.Uint64(k)
 		if n >= stopAtBlock { // [from, to)
+			if stopAtBlock == blockTo { // in order to prevent situation when we quit the cycle before reaching blockTo
+				existBlocksToPrune = false // and think that there is nothing to prune but there is
+			}
 			break
 		}
 
```

Comment:
Two main points of this PR:

    We assume that there is smth to prune by default
    The only reason to stop pruning is to reach blockTo (not blockDeleteLimit)